pFUnit directives regularly trigger S001

# The problem

Due to the implementation of pFUnit's pre-processor, directives (i.e. `@assertEqual`, `@Test`) must be defined on a single line. Therefore, they often fail the [line-too-long (S001)](https://fortitude.readthedocs.io/en/stable/rules/line-too-long/) rule.

## Possible Solution

Rule S001 could be disabled for any `custom_directives` returned by tree-sitter-fortran

## Minimal example

Using the below example...

```f90
module pfunit_long_line
    ! allow(C121)
    use funit
    implicit none (type, external)
    private

contains
    @Test
    subroutine simple_test_with_long_line_assertion()
        integer :: long_name_for_first_variable, long_name_for_second_variable

        long_name_for_first_variable = 1
        long_name_for_second_variable = 1

        @assertEqual(long_name_for_first_variable, long_name_for_second_variable, message="I cannot wrap lines within a pFUnit directive")
    end subroutine simple_test_with_long_line_assertion
end module pfunit_long_line
```

I get the following output from fortitude

```
$ fortitude check pfunit_long_line.pf
pfunit_long_line.pf:15:101: S001 line length of 138, exceeds maximum 100
   |
13 | ...
14 | ...
15 | ...ame_for_second_variable, message="I cannot wrap lines within a pFUnit directive")
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ S001
16 | ...
17 | ...
   |

fortitude: 1 files scanned.
Number of errors: 1

For more information about specific rules, run:

    fortitude explain X001,Y002,...
```

--- a/crates/fortitude_linter/resources/test/fixtures/style/S001_line_length_pfunit_ok.pf
+++ b/crates/fortitude_linter/resources/test/fixtures/style/S001_line_length_pfunit_ok.pf
@@ -1,0 +1,17 @@
+module pfunit_long_line
+    ! allow(C121)
+    use funit
+    implicit none (type, external)
+    private
+
+contains
+    @Test
+    subroutine simple_test_with_long_line_assertion()
+        integer :: long_name_for_first_variable, long_name_for_second_variable
+
+        long_name_for_first_variable = 1
+        long_name_for_second_variable = 1
+
+        @assertEqual(long_name_for_first_variable, long_name_for_second_variable, message="I cannot wrap lines within a pFUnit directive")
+    end subroutine simple_test_with_long_line_assertion
+end module pfunit_long_line

--- a/crates/fortitude_linter/src/rules/style/line_length.rs
+++ b/crates/fortitude_linter/src/rules/style/line_length.rs
@@ -34,6 +34,8 @@ use tree_sitter::Point;
 /// 3. Ignores SPDX license identifiers and copyright notices (for example, `!
 ///    SPDX-License-Identifier: MIT`), which are machine-readable and should
 ///    _not_ wrap over multiple lines.
+/// 4. Ignores pFUnit test directives, as these must be defined on a single
+///    line.
 ///
 /// Note that the Fortran standard states a maximum line length of 132
 /// characters[^1], and while most modern compilers will support longer lines[^2],
@@ -134,6 +136,12 @@ impl LineTooLong {
                 (first_chunk, second_chunk),
                 ("!", "SPDX-License-Identifier:" | "SPDX-FileCopyrightText:")
             ) {
+                continue;
+            }
+
+            // Do not enforce the line length limit for pFUnit test directives,
+            // which must be defined on a single line.
+            if first_chunk.starts_with("@") {
                 continue;
             }
 

--- a/crates/fortitude_linter/src/rules/style/mod.rs
+++ b/crates/fortitude_linter/src/rules/style/mod.rs
@@ -70,6 +70,7 @@ mod tests {
         Ok(())
     }
 
+    #[test_case(Rule::LineTooLong, Path::new("S001_line_length_pfunit_ok.pf"))]
     #[test_case(Rule::SuperfluousImplicitNone, Path::new("S201_ok.f90"))]
     #[test_case(Rule::UnsortedUses, Path::new("S271_ok.f90"))]
     fn rules_pass(rule_code: Rule, path: &Path) -> Result<()> {

--- a/docs/rules/line-too-long.md
+++ b/docs/rules/line-too-long.md
@@ -26,6 +26,8 @@ determining whether a line is overlong. Namely, it:
 3. Ignores SPDX license identifiers and copyright notices (for example, `!
    SPDX-License-Identifier: MIT`), which are machine-readable and should
    _not_ wrap over multiple lines.
+4. Ignores pFUnit test directives, as these must be defined on a single
+   line.
 
 Note that the Fortran standard states a maximum line length of 132
 characters[^1], and while most modern compilers will support longer lines[^2],

